src/app/aws/aws-parameters.ts

```diff
@@ -47,16 +47,18 @@ const getAllParams = function getGuardianConfigurationRecursiveStep(
 	params = [],
 	token: string | undefined = undefined,
 ): Promise<AWSParameter[]> {
-	return getParams(stage, token).then((response) => {
-		if (!response.NextToken) {
-			return params;
-		}
-		return getAllParams(
-			stage,
-			params.concat(response.Parameters),
-			response.NextToken === '' ? undefined : response.NextToken,
-		);
-	});
+	return getParams(stage, token).then(
+		(response: Response & { NextToken?: any; Parameters?: any }) => {
```

Comment:
I am assuming that what is returned is a `Response` type with some AWS caviats `{ NextToken?: any; Parameters?: any }`

Reply:
Parameters can be narrowed to `AWSParameter[]` here I think.

Reply:
Getting really specific, we could just use the AWS types here

```
import { Parameter } from 'aws-sdk/clients/ssm';
import { PromiseResult } from 'aws-sdk/lib/request';

// ...

const getParams = function getAWSParameterStoreParameters(
	stage: string,
	token: string | undefined = undefined,
): Promise<PromiseResult<AWS.SSM.GetParametersByPathResult, AWS.AWSError>> {
	const params = {
		Path: `/${STACK}/${stage}/`,
		Recursive: true,
		WithDecryption: true,
		NextToken: token,
	};

	return ssm.getParametersByPath(params).promise();
};

// a recursive function to retrieve all pages of guardian configuration

const getAllParams = function getGuardianConfigurationRecursiveStep(
	stage: string,
	params: Parameter[] = [],
	token: string | undefined = undefined,
): Promise<Parameter[]> {
	return getParams(stage, token).then((response) => {
		if (!response.NextToken) {
			return params;
		}
		return getAllParams(
			stage,
			params.concat(response.Parameters || []),
			response.NextToken === '' ? undefined : response.NextToken,
		);
	});
};```